test/e2e/namespace_annotation_test.go

```diff
@@ -0,0 +1,122 @@
+			RunNamespaceAnnotationTest(fiveMinTimeout, client, "nstest-"+uuidgen.String(), 2,
+				backupName, restoreName)
+		})
+	})
+})
+
+func RunNamespaceAnnotationTest(ctx context.Context, client testClient, nsBaseName string, numberOfNamespaces int, backupName string, restoreName string) error {
+	//timeout logic
+	shortTimeout, _ := context.WithTimeout(ctx, 5*time.Minute)
+
+	//defer does not run until the actual surrounding function returns a value, this line is meant for post test cleanup
+	defer cleanupNamespaces(ctx, client, nsBaseName) // Run at exit for final cleanup
+	var excludeNamespaces []string
+
+	// Currently it's hard to build a large list of namespaces to include and wildcards do not work so instead
+	// we will exclude all of the namespaces that existed prior to the test from the backup
+	namespaces, err := client.clientGo.CoreV1().Namespaces().List(shortTimeout, v1.ListOptions{})
+	if err != nil {
+		return errors.Wrap(err, "Could not retrieve namespaces")
+	}
+
+	//appending current namespaces to list of excluded namespaces, and no operations will be performed on those
+	for _, excludeNamespace := range namespaces.Items {
+		excludeNamespaces = append(excludeNamespaces, excludeNamespace.Name)
+	}
+
+	//creates the test namespaces mentioned in arguments of the function, with format of nstest-someuuid. This loops runs up to the number mentioned in the argument
+	for nsNum := 0; nsNum < numberOfNamespaces; nsNum++ {
+		createNSName := fmt.Sprintf("%s-%00000d", nsBaseName, nsNum)
+		createAnnotationName := fmt.Sprintf("annotation-%s-%00000d", nsBaseName, nsNum)
+		if err := createNamespaceAndAnnotation(ctx, client, createNSName, createAnnotationName); err != nil {
+			return errors.Wrapf(err, "Failed to create namespace %s", createNSName)
+		}
+	}
+
+	if err := veleroBackupExcludeNamespaces(ctx, veleroCLI, veleroNamespace, backupName, excludeNamespaces); err != nil {
+		veleroBackupLogs(ctx, veleroCLI, "", backupName)
+		return errors.Wrapf(err, "Failed to backup backup namespaces %s-*", nsBaseName)
+	}
+
+	//if error in cleaning up namespaces, then throw error
+	err = cleanupNamespaces(ctx, client, nsBaseName)
+	if err != nil {
+		return errors.Wrap(err, "Could cleanup retrieve namespaces")
+	}
+
+	//if error in restoring namespaces, then throw error
+	err = veleroRestore(ctx, veleroCLI, veleroNamespace, restoreName, backupName)
+	if err != nil {
+		return errors.Wrap(err, "Restore failed")
+	}
+
+	// Verify that we got back all of the namespaces we created, (and also check if we got back the annotations)
+	for nsNum := 0; nsNum < numberOfNamespaces; nsNum++ {
+		checkNSName := fmt.Sprintf("%s-%00000d", nsBaseName, nsNum)
+		checkAnnoName := fmt.Sprintf("annotation-%s-%00000d", nsBaseName, nsNum)
+		checkNS, err := getNamespace(shortTimeout, client, checkNSName)
+		c := checkNS.ObjectMeta.Annotations["testAnnotation"]
+
+		if err != nil {
+			return errors.Wrapf(err, "Could not retrieve test namespace %s", checkNSName)
+		}
```

Comment:
Check for error before trying to access annotations.

Reply:
added changes